snakemake/dag.py

```diff
@@ -735,6 +735,11 @@ def update(
         exceptions = list()
         cycles = list()
 
+        # check if all potential producers are strictly ordered
+        jobs = sorted(jobs, reverse=True)
+        primary_job = jobs[0]
+        is_strictly_ordered = all(primary_job > job for job in jobs)
```

Comment:
This is not correct, as the first element shall be compared to the rest of the elements:
```python
is_strictly_ordered = all(primary_job > job for job in jobs[1:])
```

Reply:
good catch!